### PR TITLE
Feature/refactor db models and fix some bugs 

### PR DESCRIPTION
**Pull request by quanpython**

### CODE REVIEW

File: studio/app/common/routers/workspace.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def search_workspaces(
     db: Session = Depends(get_db),
     current_user: User = Depends(get_current_user),
 ):
```
Comment (itutu-tienday) [inline studio/app/common/routers/workspace.py:34]: @quanpython 

Please add the following fields to the model Workspace
- shared_count (number of shared users)

\* The above information is reflected in the API documentation v0.1.7.
- https://drive.google.com/drive/folders/1vCyE0B8kMcZZgymXMnVt-_PYMRlHD20a
  - optinist-server-apidoc-v0.1.7.html

**Review by itutu-tienday — CHANGES_REQUESTED** [review]

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]
apply same changes (add relationship and FK) for all models.

- User
  - organization
- UserRole
  - user
  - role
- WorkspaceShareUser
  - workspace
  - user
- Experiment
  - organization
- ExperimentShareUser
  - experiment
  - user

File: studio/app/common/routers/workspace.py (hunk 1 above)
Comment (itutu-tienday) [reply studio/app/common/routers/workspace.py:43]: @quanpython 
The above seems to have been addressed in 083f68d, but we have identified the following issue.

- In some cases, the number of user lists obtained by `GET /workspace/share/{id}/status` and shared_count do not match.
  - -> Probably due to the difference in flag judgments such as active in the users table.

Please check and adjust the above.

**Review by itutu-tienday — CHANGES_REQUESTED** [review]

File: studio/app/common/routers/workspace.py (hunk 1 above)
Comment (itutu-tienday) [reply studio/app/common/routers/workspace.py:51]: Note: Correction confirmed at 671087c.

**Review by itutu-tienday — APPROVED** [review]

**Review by ReiHashimoto — APPROVED** [review]
LGTM

**Pull request merged**